File: pharmacie/src/main/java/io/pharmacie/SignupActivity.java
package io.pharmacie;

import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.support.design.widget.TextInputEditText;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;
import butterknife.BindView;
import butterknife.ButterKnife;
import io.pharmacie.Retrofit.Api;
import io.pharmacie.Retrofit.RetrofitClient;
import io.pharmacie.models.User;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class SignupActivity extends AppCompatActivity {

    @BindView(R.id.link_login) TextView _loginLink;
    @BindView(R.id.input_firstname)TextInputEditText _input_firstname;
    @BindView(R.id.input_lastname) TextInputEditText _input_lastname;
    @BindView(R.id.input_email) TextInputEditText _input_email;
    @BindView(R.id.input_mobile) TextInputEditText _input_mobile;
    @BindView(R.id.input_NSS) TextInputEditText _input_NSS;
    @BindView(R.id.btn_signup) Button _signupButton;

    final String  accountSid= "AC611f2652a8f2262f662a0308bb9f7cd5";
    final String authToken ="7cd41eb1ae681c57c89c0c5178f135de";
    final String fromPhoneNumber = "+12012318756";
    final int numbterLetterPwd = 3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);
        ButterKnife.bind(this);

        View decorView = getWindow().getDecorView();
        MainActivity.hideSystemUI(decorView);

        _signupButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                User user = new User(_input_lastname.getText().toString(),_input_firstname.getText().toString(),_input_email.getText().toString()
                        ,_input_mobile.getText().toString(),_input_NSS.getText().toString(),generatePwd(6));

                Retrofit.Builder builder = new Retrofit.Builder()
                        .baseUrl(Api.Base_Url)
                        .addConverterFactory(GsonConverterFactory.create());
                Retrofit retrofit = builder.build();

                Api client = retrofit.create(Api.class);
                Call<User> call = client.createAccount(user);
                call.enqueue(new Callback<User>() {
                    @Override
                    public void onResponse(Call<User> call, Response<User> response) {

                        Toast.makeText(SignupActivity.this,"something went great :)",Toast.LENGTH_SHORT).show();

                    }

                    @Override
                    public void onFailure(Call<User> call, Throwable t) {
                       Toast.makeText(SignupActivity.this,"something went wrong :(",Toast.LENGTH_SHORT).show();
                    }
                });



            }
        });


        _loginLink.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Finish the registration screen and return to the Login activity
                MainActivity.PreviousClass = SignupActivity.class;
                Intent intent = new Intent(getApplicationContext(),LoginActivity.class);
                startActivity(intent);
                finish();
                overridePendingTransition(R.anim.push_left_in, R.anim.push_left_out);
            }
        });
    }

    public String generatePwd(int length) {
        String chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
        String pass = "";
        for(int x=0;x<length;x++)
        {
            int i = (int)Math.floor(Math.random() * 62);
            pass += chars.charAt(i);
        }
        System.out.println(pass);
        return pass;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event)
    {
        if (keyCode == KeyEvent.KEYCODE_BACK)
        {
            Intent intentprevious = new Intent(getApplicationContext(),MainActivity.PreviousClass);
            startActivity(intentprevious);
            finish();
            overridePendingTransition(R.anim.push_left_in, R.anim.push_left_out);
        }

        if (keyCode == KeyEvent.KEYCODE_HOME)
        {

        }

        // // TODO Auto-generated method stub
        return super.onKeyDown(keyCode, event);

    }
}

File: pharmacie/src/main/java/io/pharmacie/SignupActivity.kt
package io.pharmacie

import android.content.Context
import android.content.Intent

import com.google.android.material.textfield.TextInputEditText
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.KeyEvent
import android.view.View
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import butterknife.BindView
import butterknife.ButterKnife
import io.pharmacie.Retrofit.Api
import io.pharmacie.Retrofit.ResponseMessage
import io.pharmacie.models.User
import kotlinx.android.synthetic.main.activity_login.*
import kotlinx.android.synthetic.main.activity_signup.*
import kotlinx.android.synthetic.main.activity_signup.input_email
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class SignupActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_signup)
        ButterKnife.bind(this)

        val decorView = window.decorView
        MainActivity.hideSystemUI(decorView)

        val code = generatePwd(4)

        btn_signup!!.setOnClickListener {
            val user = User(
                input_lastname!!.text!!.toString(),
                input_firstname!!.text!!.toString(),
                input_email!!.text!!.toString(),
                input_mobile!!.text!!.toString(),
                input_NSS!!.text!!.toString(),code
                //generatePwd(6)
            )

            val builder = Retrofit.Builder()
                .baseUrl(Api.Base_Url)
                .addConverterFactory(GsonConverterFactory.create())
            val retrofit = builder.build()

            val client = retrofit.create(Api::class.java)
            val call = client.createAccount(user)
            call.enqueue(object : Callback<ResponseMessage> {
                override fun onResponse(call: Call<ResponseMessage>, response: Response<ResponseMessage>) {

                    val repense = response.body()
                    val result = repense?.result
                    if(result != null)
                    if(result!!) {
                        val pref = getSharedPreferences("ahlamfile", Context.MODE_PRIVATE)
                        with(pref.edit()){
                            putBoolean("connected", true).commit()
                            putString("email",  input_email!!.text!!.toString()).commit()
                            putString("pwd", code).commit()
                        }
                        Toast.makeText(this@SignupActivity, repense?.message.toString(), Toast.LENGTH_SHORT).show()
                        val intent = Intent(applicationContext, ConfirmationActivity::class.java)
                        startActivity(intent)
                        finish()
                        overridePendingTransition(R.anim.push_left_in, R.anim.push_left_out)
                    }else{
                        Toast.makeText(this@SignupActivity, repense?.message.toString(), Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<ResponseMessage>, t: Throwable) {
                    Toast.makeText(this@SignupActivity, "something went wrong :(", Toast.LENGTH_SHORT).show()
                }
            })
        }


        link_login!!.setOnClickListener {
            // Finish the registration screen and return to the Login activity
            MainActivity.PreviousClass = SignupActivity::class.java
            val intent = Intent(applicationContext, LoginActivity::class.java)
            startActivity(intent)
            finish()
            overridePendingTransition(R.anim.push_left_in, R.anim.push_left_out)
        }
    }

    fun generatePwd(length: Int): String {
        val chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890"
        var pass = ""
        for (x in 0 until length) {
            val i = Math.floor(Math.random() * 62).toInt()
            pass += chars[i]
        }
        println(pass)
        return pass
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean {
        if (keyCode == KeyEvent.KEYCODE_BACK) {
            val intentprevious = Intent(applicationContext,  MainActivity::class.java)
            startActivity(intentprevious)
            finish()
            overridePendingTransition(R.anim.push_left_in, R.anim.push_left_out)
        }

        if (keyCode == KeyEvent.KEYCODE_HOME) {

        }

        // // TODO Auto-generated method stub
        return super.onKeyDown(keyCode, event)

    }

}